### PR TITLE
Fix DB meta lost bug

### PR DESCRIPTION
**Pull request by kangkaisen**
For https://github.com/apache/incubator-doris/issues/1964

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 1]
```diff
@@ -5738,11 +5739,17 @@ public long loadCluster(DataInputStream dis, long checksum) throws IOException,
...
+                if (!fullNameToDb.containsKey(dbName)) {
+                    final InfoSchemaDb db = new InfoSchemaDb(cluster.getName());
+                    db.setClusterName(cluster.getName());
+                    String errMsg = "InfoSchemaDb id shouldn't larger than 10000, please restart your FE server";
+                    // ensure InfoSchemaDb meta don't overwrite the normal db meta
+                    Preconditions.checkState(db.getId() < NEXT_ID_INIT_VALUE, errMsg);
+                    idToDb.put(db.getId(), db);
+                    fullNameToDb.put(db.getFullName(), db);
+                }
+                cluster.addDb(dbName, fullNameToDb.get(dbName).getId());
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java]: I'm not sure why this change can fix database lost, could you please add some comment to explain it?

**Comment by kangkaisen** [comment]
1 when checkpoint, every time `loadCluster`, we will `new InfoSchemaDb`, the id for `InfoSchemaDb` will more and more larger，and finally will larger than 10000
2 when InfoSchemaDb id larger than 10000 and call `idToDb.put(db.getId(), db)`, the InfoSchemaDb meta will overwrite the  normal db meta, (such as, the  normal db id is 10001),so the normal db meta will lost when checkpoint generate the image file by `saveDb`.

**Comment by imay** [comment]
> 1 when checkpoint, every time `loadCluster`, we will `new InfoSchemaDb`, the id for `InfoSchemaDb` will more and more larger，and finally will larger than 10000
> 2 when InfoSchemaDb id larger than 10000 and call `idToDb.put(db.getId(), db)`, the InfoSchemaDb meta will overwrite the normal db meta, (such as, the normal db id is 10001),so the normal db meta will lost when checkpoint generate the image file by `saveDb`.

@kangkaisen 
Thanks for your detailed explanation. Could you please add this explaination to the code in order to make it clear for others?

**Comment by kangkaisen** [comment]
> Thanks for your detailed explanation. Could you please add this explaination to the code in order to make it clear for others?

@imay Update comment.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 1 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: How do you guarantee that the InfoSchemaDb you created here has id less than NEXT_ID_INIT_VALUE?
And I think since you've already changed the `saveDb()` method, which will skip InfoSchemaDb by name, not by ID. Here we do not need to check the ID anymore.

Comment (morningman) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: Oh, I see. The id of InfoSchemaDb is under another id serial.
So the problem is that if we create more than 10000 clusters, this may still be a problem?

Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: > So the problem is that if we create more than 10000 clusters, this may still be a problem? However it is almost impossible to happen.

Yes. I also considered this problem.  I think cluster will be obsolete after https://github.com/apache/incubator-doris/issues/1723, so I don't add a check when create cluster.

If you think the check is necessary, I could add a check.

**Review by imay — APPROVED** [review]
LGTM

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 1 above)
Comment (morningman) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java]: It' OK. Don' t have to change~
